provider/pkg/schema/gen.go

```diff
@@ -1225,6 +1225,12 @@ func (ctx *cfSchemaContext) genEnumType(enumName string, propSchema *jsschema.Sc
 	case "networkfirewall:RuleGroupType":
 		typName = "RuleGroupTypeEnum" // Go SDK name conflict vs. RuleGroup resource
 	}
+
+	switch ctx.cfTypeName + "   " + enumName {
+	case "AWS::QBusiness::Index   Status":
+		typName = "QBusinessIndexStatus"
```

Comment:
This could happen at any point again, right? Is there a generic approach we could implement to prevent this from happening again in the future? I guess namespacing (like adding the service name in front) the enum types now would be a breaking change, right?